### PR TITLE
Examples / RTFD update

### PR DESCRIPTION
**Pull request by MuellerSeb**
With this PR the following stuff is proposed:

- reworked all examples to be up to date
  - herten example now uses pyvista and comes with pre-processed date (sphinx bugfix), solving #33
  - the tutorial for the covariance model was adopted to the sphinx gallery style from #63 (thanks again @banesullivan)
  - lots of cleanups
  - used ``import gstools as gs`` everywhere
  - tried to use the builtin plotting routines as often as possible
- sphinx autosummary now creates single pages for every covariance model
- better documentation for "correlation" and "normalized" correlation function
- a ``cor`` method is now automatically added to every cov-model
- added some missing vtk routines to the ``__init__.py``, solving #47 
- some corrected relative imports in the covmodel submodule to make use of autosummary

# Question
Should we use the autosummary elsewhere in the documentation?

- [x] Yeah we should!
- [ ] No!

### CODE REVIEW

**Comment by MuellerSeb** [comment]
You can have a look at the resulting documentation here:
https://geostat-framework.readthedocs.io/projects/gstools/en/examples_update

File: docs/source/_templates/autosummary/class.rst [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
+.. currentmodule:: {{ module }}
+
+.. autoclass:: {{ objname }}
    :members:
    :undoc-members:
-   :show-inheritance:
    :inherited-members:
+   :show-inheritance:
 
 .. raw:: latex
```
Comment (MuellerSeb) [inline docs/source/_templates/autosummary/class.rst]: Sometimes git tries to do its best to interpret changes but fails ungracefully.

File: README.md [hunk 2]
```diff
@@ -113,12 +113,12 @@ srf.plot()
 A similar example but for a three dimensional field is exported to a [VTK](https://vtk.org/) file, which can be visualized with [ParaView](https://www.paraview.org/) or [PyVista](https://docs.pyvista.org) in Python:
 
 ```python
-from gstools import SRF, Gaussian
 import matplotlib.pyplot as pt
```
Comment (MuellerSeb) [inline README.md]: Could we stick to ``plt`` as used in the whole matplotlib documentation?

File: docs/source/_templates/autosummary/class.rst (hunk 1 above)
Comment (LSchueler) [reply docs/source/_templates/autosummary/class.rst]: What a git!

File: README.md (hunk 2 above)
Comment (MuellerSeb) [reply README.md]: Why is matplotlib even imported here? We could kick it out of the README almost everywhere.

File: examples/01_random_field/01_srf_ensemble.py [hunk 3]
```diff
@@ -46,6 +46,7 @@
 
 
 from gstools.random import MasterRNG
```
Comment (LSchueler) [inline examples/01_random_field/01_srf_ensemble.py]: `import gstools as gs`?

File: docs/source/index.rst [hunk 4]
```diff
@@ -118,12 +118,12 @@ with a :any:`Gaussian` covariance model.
 
 .. code-block:: python
 
-    from gstools import SRF, Gaussian
     import matplotlib.pyplot as plt
```
Comment (MuellerSeb) [inline docs/source/index.rst]: matplotlib not used here.

File: examples/01_random_field/03_unstr_srf_export.py [hunk 5]
```diff
@@ -7,30 +7,25 @@
 then create a random field at those coordinates.
 """
 import numpy as np
-import matplotlib.pyplot as pt
-from gstools import SRF, Exponential
-from gstools.random import MasterRNG
```
Comment (LSchueler) [inline examples/01_random_field/03_unstr_srf_export.py]: Should `MasterRNG` actually be in the `gstools` namespace? - Not so much going on in `gstools.random` and `MasterRNG` is self-explanatory enough to not confuse people.

File: examples/03_variogram/01_variogram_estimation.py [hunk 6]
```diff
@@ -8,8 +8,6 @@
 this example script.
 
 
-*This example will only work with Python 3.*
```
Comment (LSchueler) [inline examples/03_variogram/01_variogram_estimation.py]: The Python 2 stuff is still all over the place... :-)

File: examples/01_random_field/04_srf_merge.py [hunk 7]
```diff
@@ -7,39 +7,32 @@
...
 y2 = rng.randint(20, 80, size=10000)
 
 field2 = srf((x2, y2))
-
-pt.tricontourf(x, y, field.T)
-pt.tricontourf(x2, y2, field2.T)
-pt.axes().set_aspect('equal')
-pt.show()
+ax.tricontourf(x2, y2, field2.T)
+ax.set_aspect("equal")
```
Comment (LSchueler) [inline examples/01_random_field/04_srf_merge.py]: `show()` this stuff

File: docs/source/index.rst (hunk 4 above)
Comment (LSchueler) [reply docs/source/index.rst]: You're right! - I just deleted all the unnecessary imports.

File: README.md (hunk 2 above)
Comment (LSchueler) [reply README.md]: done

**Pull request merged**